### ISSUE TITLE
Building packages with `package-build.el` does not work on Windows.

### ISSUE THREAD

**Issue by jaccarmac** [milkypostman/melpa#2770]
The problem stems from the addition of `env x=y` to the command string [here](https://github.com/milkypostman/melpa/blob/52a7598dc550c76f4e081fe1c4a6d8697bd30561/package-build.el#L225). Package repos will clone if the line is changed to `'()`, but then `package.el` cannot find the VERSION or PACKAGE-VERSION headers in the downloaded file(s).

**Comment by milkypostman** [comment]
if you wanted to provide a patch we would certainly consider it, but right
now there is very little motivation to make this work on windows; at least
from our perspective.

i'm not trying to be negative, but I fear that once we start to support
something like this, then we will need to invest an even larger amount of
time supporting it.

**Comment by jaccarmac** [comment]
The problem lies in the way WIndows deals with locales, as far as I can tell. My cursory investigation found that there is in fact no way to set the locale of a running program on Windows with console commands or environment variables. You have to use an API call to do it. This represents a completely different paradigm in dealing with locales when compared to POSIX, so I completely understand your position. I believe that supporting locale-editing for subprocesses on Windows would require a patch to Emacs core or a wrapper program of some kind.

I am also looking into other ways to get Emacs to behave as expected on Windows. If there's a MELPA patch in the cards, I will definitely send it in!

**Comment by milkypostman** [comment]
thanks for your understanding!

**Comment by purcell** [comment]
In this particular case it would be okay to simply not prepend the command with `env LC_ALL=C` on Windows, so a patch to that effect would be acceptable, if you can confirm that it makes things work again on that platform.

We'd like this to work on Windows, but we're just unable to test it ourselves. :-)

**Comment by jaccarmac** [comment]
Removing the `LC_ALL` line does indeed let the package get cloned, as does adding Cygwin's `env` to the Windows `PATH`. However, when I tried that, I ran into encoding issues. @steckerhalter found a few solutions in quelpa/quelpa#72. Basically, the problem boils down to line endings, and from what I gather, there may be places where `insert-file-contents` is used instead of `insert-file-contents-literally` or vice versa. I don't have the repos cloned right now, so I have to rely on GitHub for grepping and it's not working very well. I will take a look in time.
